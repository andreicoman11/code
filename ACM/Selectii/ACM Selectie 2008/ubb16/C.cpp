#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>
#include <map>
#include <list>
#include <cmath>
#include <algorithm>
#include <set>
#include <iomanip>
using namespace std;

ifstream fin ("C.in");
//ofstream fout ("C.out");

int N;

int main()
{
	string e1, e2;
	fin >> e1;
	while( e1[0]!='E' )
	{
		double h, t, d;
		h = t = d = -200;
		if( e1[0]=='T' ) fin >> t;
		if( e1[0]=='H' ) fin >> h;
		if( e1[0]=='D' ) fin >> d;

		fin >> e2;
		if( e2[0]=='T' ) fin >> t;
		if( e2[0]=='H' ) fin >> h;
		if( e2[0]=='D' ) fin >> d;
		
		if( d!=-200 )
		{
			double e = 6.11 * exp (5417.7530 * ((1/273.16) - (1/(d+273.16))));
			double A = (0.5555)*(e - 10.0);
			if( h==-200 )
				h = t + A;
			else
				t = h - A;
		}
		else
		{
			//d = ((double)1) / (-1)*(((log(((h-t)/0.5555 + 10.0)/6.11))/5417.7530 - 1/273.16)) - 273.16;
			double A = (h-t)/0.5555 + 10.0;
			A = log(A/6.11);
			A = A/5417.7530;
			A = 1/273.16 - A;
			d = ((double)1)/A - 273.16;
		}
		cout.setf(ios::fixed);
		cout << setprecision(1) << "T " << t << " D " << d << " H " << h << "\n";

		fin >> e1;
	}

	return 0;
}

