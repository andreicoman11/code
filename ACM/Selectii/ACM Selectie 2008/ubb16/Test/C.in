T 30 D 15
T 30 D 25
D 15 H 34.0
D 25 H 42.3
T 30.0 H 34.0
T 30.0 H 42.3
T 99 D -99
T 30 H 25
T 30 D -22.9
D -30 T -22.9
H 99 T 97
E
