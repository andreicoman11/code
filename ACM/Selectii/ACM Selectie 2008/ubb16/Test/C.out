T 30.0 D 15.0 H 34.0
T 30.0 D 25.0 H 42.3
T 30.0 D 15.0 H 34.0
T 30.0 D 25.0 H 42.3
T 30.0 D 15.0 H 34.0
T 30.0 D 25.0 H 42.3
T 99.0 D -99.0 H 93.4
T 30.0 D -22.9 H 25.0
T 30.0 D -22.9 H 25.0
T -22.9 D -30.0 H -28.2
T 97.0 D 11.5 H 99.0
